app-lambda/rflambda/model.py

```diff
@@ -28,6 +28,8 @@ def handler(event: eventType, context: Dict[str, Any]):
     scene_to_post = create_l8.create_scene(event) if isinstance(
         event, NewLandsat8Event) else create_s2.create_scene(event)
     logger.info('Sending scene to the Raster Foundry API at %s', api_host)
-    result = rf_api.client.Imagery.post_scenes(scene=scene_to_post).result()
+    result = rf_api.client.Imagery.post_scenes(
+        scene=scene_to_post,
+        _request_options={'headers': {'User-Agent': 'RFLambda'}}).result()
```

Comment:
Is there a way to get at the module version? I see it in `setup.py`, but not sure if that automatically makes it available via `__version__`.

See: https://stackoverflow.com/a/20180597

Adding this may also resolve https://github.com/raster-foundry/raster-foundry/issues/4639.

Reply:
Done! As I found in https://packaging.python.org/guides/single-sourcing-package-version/, there isn't one good way of maintaining a single source of truth for a package version. I decided to go with a hybrid of bullets 3 and 6: a `__version__` variable defined in `version.py`, which is imported into `__init__.py` and then read into `setup.py` during packaging. This has the disadvantage that `__init__.py` can't import any third-party modules defined in `install_requires` (since they generally won't be available when `setup.py` gets parsed for installation), but it has the advantages that A) application code can read the version number through `from version import __version__` and B) only one variable needs to get updated to bump the version.

After implementing this, I started to wonder whether it would be more useful to report the lambda function version (which we get for free in `$AWS_LAMBDA_FUNCTION_VERSION`). On the one hand, I don't feel super confident that devs will remember to update `__version__`, since this module isn't actually distributed on PyPi (e.g. I noticed that some other metadata was incorrect during this PR, which I had to fix in order to actually import the module); on the other hand, this function code seems to change so rarely that reporting the Lambda function version may erroneously imply that the code is different.

Reply:
I think the hybrid approach makes sense. [`setuptools-scm`](https://pypi.org/project/setuptools-scm/) looks promising, but probably a larger effort to integrate in this case.

The inclusion of `AWS_LAMBDA_FUNCTION_VERSION` seems like a good idea. Possibly we can do something that concatenates the two, and defaults to something reasonable when `AWS_LAMBDA_FUNCTION_VERSION` is not present?

For example:

```
0.1.0-$AWS_LAMBDA_FUNCTION_VERSION
0.1.0-dev
```

Additionally, have you seen what a value of `AWS_LAMBDA_FUNCTION_VERSION` looks like? I suspect they are integers?

Reply:
I like the idea of concatenating the two versions. Digging deeper into Lambda function versions, one snag is that we're not actually publishing new versions of these functions. Per the [Terraform docs](https://www.terraform.io/docs/providers/aws/r/lambda_function.html#publish):

> `publish` - (Optional) Whether to publish creation/change as new Lambda Function Version. Defaults to false.

We don't set this attribute in the Terraform config for these functions ([source](https://github.com/azavea/raster-foundry-deployment/blob/6037fe44e8a9a9f33e7bd659c081078d8fe0b936/core/terraform/lambda.tf#L222-L247)). I'm not quite sure how to format the version string, anyway; the [Lambda versioning docs](https://docs.aws.amazon.com/lambda/latest/dg/versioning-aliases.html) support your suspicion that they're integers, as far as I can tell. (They don't come out and say this explicitly, but in all the examples they're integers, and I don't see a way to specify the version string.)

Perhaps more useful to us would be to create an environment variable for the function using the commit SHA, and then update that variable on every deploy. E.g. 

```diff
--- a/core/terraform/lambda.tf
+++ b/core/terraform/lambda.tf
@@ -237,6 +237,7 @@ resource "aws_lambda_function" "sentinel2_sns_addition" {
       RF_REFRESH_TOKEN     = "${var.auth0_system_refresh_token}"
       ENVIRONMENT          = "${var.environment}"
       ROLLBAR_SERVER_TOKEN = "${var.rollbar_server_token}"
+      GIT_COMMIT           = "${var.image_version}"
     }
   }
```

Then we could report the User-Agent as `RFLambda ${pyversion}-${git_commit}`.